cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -89,7 +89,8 @@ public void suspendWorkflowRun(HttpRequest request, final HttpResponder responde
       Id.Program id = Id.Program.from(namespaceId, appId, ProgramType.WORKFLOW, workflowName);
       ProgramRuntimeService.RuntimeInfo runtimeInfo = runtimeService.list(id).get(RunIds.fromString(runId));
       if (runtimeInfo == null) {
-        sendInvalidResponse(responder, id);
+        responder.sendString(HttpResponseStatus.NOT_FOUND, String.format("Runtime information for run-id %s not found",
```

Comment:
I think the new standard for returning not found is to throw NotFoundException. Check with @jwang47